external-configuration-store/ExternalConfigurationStore/ExternalConfigurationStore.Cloud/SettingsStore/BlobSettingsStore.cs

```diff
@@ -1,38 +1,41 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Azure;
+using Azure.Storage.Blobs;
+using Microsoft.Azure;
+
 namespace ExternalConfigurationStore.Cloud.SettingsStore
 {
-    using System.Collections.Generic;
-    using System.IO;
-    using System.Linq;
-    using System.Threading.Tasks;
-    using System.Xml.Linq;
-    using Microsoft.Azure;
-    using Microsoft.WindowsAzure.Storage;
-    using Microsoft.WindowsAzure.Storage.Blob;
-
     public class BlobSettingsStore : ISettingsStore
     {
-        private readonly CloudBlockBlob configBlob;
+        private readonly BlobClient configBlob;
 
         public BlobSettingsStore(string environment) : this(CloudConfigurationManager.GetSetting("storageAccount"), "configuration", "configurationdata", environment)
         {
         }
 
-        public BlobSettingsStore(string storageAccount, string configContainer, string configBlobName, string environment)
+        public BlobSettingsStore(string connectionString, string configContainer, string configBlobName, string environment)
         {
-            var account = CloudStorageAccount.Parse(storageAccount);
-            var client = account.CreateCloudBlobClient();
-            var container = client.GetContainerReference(configContainer);
+            var blobServiceClient = new BlobServiceClient(connectionString);
+            var container = blobServiceClient.GetBlobContainerClient(configContainer);
+            container.CreateIfNotExists();
 
-            this.configBlob = container.GetBlockBlobReference(configBlobName + "-" + environment + ".config");
+            this.configBlob = container.GetBlobClient(configBlobName + "-" + environment + ".config");
         }
 
-        public async Task<string> GetVersionAsync()
+        public async Task<ETag> GetVersionAsync()
         {
-                await this.configBlob.FetchAttributesAsync();
+            var response = await configBlob.GetPropertiesAsync();
+            var eTag = response.Value.ETag;
```

Comment:
NIT: You could consider just doing `return response.Value.ETag` instead of assigning it to a local.